src/Services/NasaFetchService.ts

```diff
@@ -0,0 +1,35 @@
+const axios = require('axios');
+
+const API_KEY = 'AE6PQe4KnsgG31lrRRxR95iGh8X6ZTyLjHturFwr'
+const API_URL = 'https://api.nasa.gov/mars-photos/api/v1/rovers/'
+
+export async function fetch_photos(rover: string, camera: string, sol: number = 1000, paginationStart: number, paginationEnd: number) {
+    let requestArray = []
+    let data = []
+    for (let page = paginationStart; page <= paginationEnd; page++) {
+        requestArray.push(axios.get(
+            API_URL + rover + '/photos',
+            {
+                params: {
+                    api_key: API_KEY,
+                    camera: camera,
+                    sol: sol,
+                    page: page
+                }
+            }))
+    }
+    await Promise.all(requestArray).then((res: any[]) => {
+        for (let i = 0; i < res.length; i++) {
+            data.push(res[i].data);
+        }
+    });
+    return data;
+}
+
+export async function fetch_rovers() {
+    return axios.get(API_URL, {
+        params: {
+            api_key: API_KEY
+        }

```

Comment:
this bit is similar for all the get requests (only the URL differs) - you could add a separate function to be used in all the cases